src/pages/AdvancedSearch/ResultsBlock/ResultsBlock.tsx

```diff
@@ -1,101 +1,28 @@
+import { Spinner } from "@canonical/react-components";
+import { useEffect } from "react";
 
-import ModelDetailsLink from "components/ModelDetailsLink";
-import Status from "components/Status";
 import { actions as jujuActions } from "store/juju";
 import {
+  getCrossModelQueryErrors,
   getCrossModelQueryLoaded,
   getCrossModelQueryLoading,
   getCrossModelQueryResults,
 } from "store/juju/selectors";
 import { useAppDispatch, useAppSelector } from "store/store";
 
 import "./_results-block.scss";
+import CodeSnippetBlock from "../CodeSnippetBlock/CodeSnippetBlock";
 
 export enum TestId {
   LOADING = "loading",
 }
 
-enum CodeSnippetView {
-  TREE = "tree",
-  JSON = "json",
-}
-
-const DEFAULT_THEME_COLOUR = "#00000099";
-
-const THEME = {
-  scheme: "Vanilla",
-  base00: "#00000000",
-  base01: DEFAULT_THEME_COLOUR,
-  base02: DEFAULT_THEME_COLOUR,
-  base03: DEFAULT_THEME_COLOUR,
-  base04: DEFAULT_THEME_COLOUR,
-  base05: DEFAULT_THEME_COLOUR,
-  base06: DEFAULT_THEME_COLOUR,
-  base07: "#000000",
-  base08: "#a86500",
-  base09: "#c7162b",
-  base0A: DEFAULT_THEME_COLOUR,
-  base0B: "#0e811f",
-  base0C: DEFAULT_THEME_COLOUR,
-  base0D: "#000000",
-  base0E: DEFAULT_THEME_COLOUR,
-  base0F: DEFAULT_THEME_COLOUR,
-};
-
-const labelRenderer: LabelRenderer = (keyPath) => {
-  const currentKey = keyPath[0];
-  // If this is a top level key then it is a model UUID.
-  if (keyPath.length === 1) {
-    return (
-      <ModelDetailsLink
-        // Prevent toggling the object when the link is clicked.
-        onClick={(event) => event.stopPropagation()}
-        title={`UUID: ${currentKey}`}
-        uuid={currentKey.toString()}
-      >
-        {currentKey}:
-      </ModelDetailsLink>
-    );
-  }
-  return <>{currentKey}:</>;
-};
-
-const valueRenderer: ValueRenderer = (valueAsString, value, ...keyPath) => {
-  const currentKey = keyPath[0];
-  const parentKey = keyPath.length >= 2 ? keyPath[1] : null;
-  // Match a status of the following structure:
-  //   "application-status": {
-  //     "current": "unknown",
-  //     ...
-  //   }
-  if (
-    currentKey === "current" &&
-    typeof parentKey === "string" &&
-    parentKey.endsWith("-status") &&
-    typeof value === "string" &&
-    typeof valueAsString === "string"
-  ) {
-    return <Status status={value}>{valueAsString}</Status>;
-  }
-  return <>{valueAsString}</>;
-};
-
 const ResultsBlock = (): JSX.Element | null => {
   const dispatch = useAppDispatch();
   const isCrossModelQueryLoading = useAppSelector(getCrossModelQueryLoading);
   const isCrossModelQueryLoaded = useAppSelector(getCrossModelQueryLoaded);
   const crossModelQueryResults = useAppSelector(getCrossModelQueryResults);
-  const [codeSnippetView, setCodeSnippetView] = useState<CodeSnippetView>(
-    CodeSnippetView.TREE
-  );
+  const crossModelQueryErrors = useAppSelector(getCrossModelQueryErrors);
   const resultsJSON = JSON.stringify(crossModelQueryResults, null, 2);
```

Comment:
I think we can move this stringify into `CodeSnippetBlock` and then we only need to pass the results or errors in.

Reply:
That makes so much sense! I'll modify it in my following commit.